### ISSUE TITLE
masking with batch size > 1 - how does this work?

### ISSUE THREAD

**Issue by sedol1339** [keras-team/keras#14338]
Let we have a batch of 2 examples of 4 timesteps, where masking is different in both examples:

example1: [unmasked, masked, masked, unmasked]
example2: [unmasked, unmasked, masked, masked]

As stated in guide, recurrent layer will ignore masked sequence elements. So layer will ignore 2-nd element in 1-st example but not in 2-nd example. But batch is meant to be passed through layers as one whole. So layer will have to split a batch into examples and pass each example individually, otherwise it cannot ignore some timesteps in specific examples.

If so, what is the point of using batch size > 1? It is generally used to speed up training by using paralellism, but if batch is splitted back, we will see degrading performance, as if we used batch_size = 1. It this correct?

**Comment by No41Name** [comment]
Any update on that?

**Comment by sushreebarsa** [comment]
@sedol1339 Sorry for the late response!
If this issue is still valid ,please provide a code snippet to reproduce the issue reported here. 
Please refer[ this](https://stackoverflow.com/questions/51789338/what-is-the-equivalence-of-masking-keras-function-in-tensorflow-and-does-batc) thread and let us know if it helps?Thanks!
